.github/workflows/CI.yml

```diff
@@ -45,6 +46,26 @@ jobs:
           command: fmt
           args: --all -- --check
 
+  clippy:
+    name: Static Code Analysis ( Clippy )
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v1
+
+      - name: Install Rust
+        uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: stable
+          override: true
+          components: clippy
+
+      - name: Clippy Check
+        uses: actions-rs/cargo@v1
+        with:
+          command: clippy
+
```

Comment:
I was thinking we could add an `args: -- -Aclippy::all -Wclippy::specific_lint_name`, and thus not need to change the actual source code (like the fmt job above).

Reply:
I have reverted the change to the main lib file , verified the build is green ! -> https://github.com/mastrzyz/hyperclip/actions/runs/3110424028/jobs/5041594746

Reply:
> My concern is something will fail in CI and not locally .

I agree that changing the top level module would prevent this, but since we're allowing everything except `self_named_module_files`, this should be fine.

Although now that I think about it, changing the top level module might not be a bad idea at all 😄

Reply:
Shouldn't we make it `-D`? We want the CI job to fail, not just yell and then get forgotten.

(I would expect this to trigger a complaint about the `ext.rs` module.)

Reply:
Sorry I misread -> Updated.


I verified that -D does in fact "Disallow"

<img width="868" alt="image" src="https://user-images.githubusercontent.com/37447884/193738966-79893f86-0229-43a0-b2f2-d26b7551368a.png">


But for the clippy rule we decided on enabling, there isn't any violation reported in ext.rs :/